refactor(curriculum): sorting songs functionality in music player project

jdwilkin4 opened the pull request:
Checklist:

<!-- Please follow this checklist and put an x in each of the boxes, like this: [x]. It will ensure that our team takes your pull request seriously. -->

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

<!--If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.-->

Closes #53553

<!-- Feel free to add any additional description of changes below this line -->

jdwilkin4 committed:
feat: adding new step 21

jdwilkin4 committed:
chore: adding new step to update function call

jdwilkin4 committed:
chore: updating seed code

jdwilkin4 committed:
chore: updating seed code for sort songs function

jdwilkin4 committed:
chore: updating seed code for sort function

jdwilkin4 committed:
fix: tests

jdwilkin4 committed:
Merge branch 'main' into refactor-53553-sorting-songs-music-player

Ksound22 commented:
I found an issue with the `sortSong` function – calling the `renderSongs` function with it (`renderSongs(sortSongs())`) does not display the songs.

In step 26, the songs were still displayed:
![Screenshot 2024-02-16 at 8 00 15 AM](https://github.com/freeCodeCamp/freeCodeCamp/assets/65571316/a4df8ec8-960d-47e3-8b7f-f063d3587e02)

By the time we had the `renderSongs(sortSongs())` call in step 27, the songs were not displayed:
![Screenshot 2024-02-16 at 8 01 27 AM](https://github.com/freeCodeCamp/freeCodeCamp/assets/65571316/987f5df2-768f-4cd4-a829-359a03836409)

In step 35, the songs were displayed too and that's because we had `renderSongs(userData?.songs)` instead of `renderSongs(sortSongs())`:
![Screenshot 2024-02-16 at 8 02 48 AM](https://github.com/freeCodeCamp/freeCodeCamp/assets/65571316/41cda6f8-543e-44e6-a5a1-657c9aecb7d3)

Changing the call back to `renderSongs(sortSongs())` did not display the songs:
![Screenshot 2024-02-16 at 8 04 59 AM](https://github.com/freeCodeCamp/freeCodeCamp/assets/65571316/968cdf1f-8b63-4829-ab58-3850013befa4)

In step 79, the songs were displayed for the same reason they were displayed in step 35:
![Screenshot 2024-02-16 at 8 08 26 AM](https://github.com/freeCodeCamp/freeCodeCamp/assets/65571316/46252ce9-e04d-4b03-b7f2-1e93943084f3)

When the preview of what to build was displayed in step 1, the songs were not displayed:
![Screenshot 2024-02-16 at 8 09 13 AM](https://github.com/freeCodeCamp/freeCodeCamp/assets/65571316/cc523b76-8cc7-4da9-a691-89e251ef606a)

Ksound22 commented:
I found an issue with the `sortSong` function – calling the `renderSongs` function with it (`renderSongs(sortSongs())`) does not display the songs.

In step 26, the songs were still displayed:
![Screenshot 2024-02-16 at 8 00 15 AM](https://github.com/freeCodeCamp/freeCodeCamp/assets/65571316/a4df8ec8-960d-47e3-8b7f-f063d3587e02)

By the time we had the `renderSongs(sortSongs())` call in step 27, the songs were not displayed:
![Screenshot 2024-02-16 at 8 01 27 AM](https://github.com/freeCodeCamp/freeCodeCamp/assets/65571316/987f5df2-768f-4cd4-a829-359a03836409)

In step 35, the songs were displayed too and that's because we had `renderSongs(userData?.songs)` instead of `renderSongs(sortSongs())`:
![Screenshot 2024-02-16 at 8 02 48 AM](https://github.com/freeCodeCamp/freeCodeCamp/assets/65571316/41cda6f8-543e-44e6-a5a1-657c9aecb7d3)

Changing the call back to `renderSongs(sortSongs())` did not display the songs:
![Screenshot 2024-02-16 at 8 04 59 AM](https://github.com/freeCodeCamp/freeCodeCamp/assets/65571316/968cdf1f-8b63-4829-ab58-3850013befa4)

In step 79, the songs were displayed for the same reason they were displayed in step 35:
![Screenshot 2024-02-16 at 8 08 26 AM](https://github.com/freeCodeCamp/freeCodeCamp/assets/65571316/46252ce9-e04d-4b03-b7f2-1e93943084f3)

When the preview of what to build was displayed in step 1, the songs were not displayed:
![Screenshot 2024-02-16 at 8 09 13 AM](https://github.com/freeCodeCamp/freeCodeCamp/assets/65571316/cc523b76-8cc7-4da9-a691-89e251ef606a)

jdwilkin4 committed:
fix: missing return step

jdwilkin4 committed:
fix: seed code for all sorting functions

jdwilkin4 committed:
fix: typo

jdwilkin4 committed:
fix: updating seed code to correct function call

jdwilkin4 commented:
Hey @Ksound22 ! 

Thanks for catching that.
The sorting function was missing the return value so I added a step for that and updated the seed code.
I also updated the seed code for the steps that were using the old `renderSongs(userData?.songs)` to replace it with the new function.

Tested it locally and we should be good now 👍

jdwilkin4 committed:
Merge branch 'main' into refactor-53553-sorting-songs-music-player
